docs/rules/jsx-pascal-case.md

```diff
@@ -0,0 +1,37 @@
+# Enforce PasalCase for user-defined JSX components (jsx-pascal-case)
+
+Enforces coding style that user-defined JSX components are defined and referenced in PascalCase.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```js
+<testComponent />
+```
+
+```js
+<testComponent>
+  <div />
+</testComponent>
+```
+
+```js
+<test_component />
+```
+
+The following patterns are not considered warnings:
+
+```js
+<TestComponent />
+```
+
+```js
+<TestComponent>
+  <div />
+</TestComponent>
+```
+
+## When Not To Use It
+
+If you are not using JSX, or if you don't want your JSX components defined and referenced in PascalCase.
```

Comment:
React will only work if user-land components are defined in PascalCase, otherwise it converts them to strings:

```
<foo />
<Foo />
```

becomes
```
  React.createElement("foo", null)
  React.createElement(Foo, null)
```

so perhaps this should just say

```
If you are not using JSX.
```